lib/rubocop/result_cache.rb

```diff
@@ -79,7 +79,12 @@ def self.cleanup(config_store, verbose, cache_root = nil)
 
     def self.cache_root(config_store)
       root = config_store.for('.')['AllCops']['CacheRootDirectory']
-      root = File.join(Dir.tmpdir, Etc.getlogin) if root == '/tmp'
+      if Etc.getlogin.nil? || Etc.getlogin.empty?
```

Comment:
Let's just simplify this to using the UID and be done with it. We've wasted enough time already.